### ISSUE TITLE
Issue 004: coarse infer config に global_path_tracing block を追加する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#353]
# Issue 004: coarse infer config に global_path_tracing block を追加する

## 背景

MVP GPT は coarse inference の optional 後処理として導入する。既存 coarse => physics => fine を壊さないため、default は後方互換の `enabled: false` とし、example / experiment YAML で明示的に有効化する。

GPT の入力 score は `log_softmax` に固定する。config で logits/probability を選べるようにはしない。

## 目的

`examples/config_infer_fbpick_coarse.yaml` と `coarse/config.py` に `infer.global_path_tracing` block を追加し、typed config として parse / validate できるようにする。

## 非目的

この issue では以下は行わない。

- DP2D の実行
- coarse/infer.py への接続
- `.coarse.npz` diagnostic key の保存
- physics / fine の変更

## 変更対象

```text
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/config.py
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/__init__.py
packages/seisai-engine/tests/test_fbpick_coarse.py
examples/config_infer_fbpick_coarse.yaml
```

必要に応じて `packages/seisai-engine/tests/test_fbpick_coarse_config.py` を新設してもよい。

## config 仕様

追加 block:

```yaml
infer:
  global_path_tracing:
    enabled: true
    algorithm: dp2d
    score_type: log_softmax
    topology: segment_1d
    max_jump_coarse_samples: 4
    min_component_anchors: 4
    fallback_short_component_to_argmax: true
    diagnostics: true
```

後方互換 default:

```yaml
infer:
  global_path_tracing:
    enabled: false
    algorithm: dp2d
    score_type: log_softmax
    topology: segment_1d
    max_jump_coarse_samples: 4
    min_component_anchors: 4
    fallback_short_component_to_argmax: true
    diagnostics: true
```

`infer.global_path_tracing` block 自体がない場合も、この default を使う。

## dataclass 仕様

`coarse/config.py` で path_tracing の `GlobalPathTracingCfg` を使うか、coarse-specific wrapper を作る。推奨は共通 dataclass を再利用すること。

```python
from seisai_engine.pipelines.fbpick.path_tracing import GlobalPathTracingCfg
```

`CoarseInferRuntimeCfg` に追加:

```python
@dataclass(frozen=True)
class CoarseInferRuntimeCfg:
    batch_size: int
    num_workers: int
    amp: bool
    use_tqdm: bool
    channel_tiling: CoarseInferChannelTilingCfg
    global_path_tracing: GlobalPathTracingCfg
```

## validation

- `enabled`: bool。
- `algorithm`: string。MVP で実行可能なのは `dp2d`。将来予約として `dp3d_crossline_smooth_then_inline` は parse してよいが、coarse infer 実行時は NotImplementedError でよい。
- `score_type`: 必ず `log_softmax`。他値は reject。
- `topology`: MVP は必ず `segment_1d`。他値は reject または 3D reserved として明確に `NotImplementedError` 対象にする。
- `max_jump_coarse_samples`: bool ではない int, `>= 0`。
- `min_component_anchors`: bool ではない int, `>= 1`。
- `fallback_short_component_to_argmax`: bool。
- `diagnostics`: bool。

## example YAML

`examples/config_infer_fbpick_coarse.yaml` に block を追加する。

推奨は、例として有効化した状態を示す。

```yaml
infer:
  global_path_tracing:
    enabled: true
    algorithm: dp2d
    score_type: log_softmax
    topology: segment_1d
    max_jump_coarse_samples: 4
    min_component_anchors: 4
    fallback_short_component_to_argmax: true
    diagnostics: true
```

ただし parser default は `enabled: false` のままにして、既存 YAML を壊さない。

## テスト要件

追加 / 更新:

```python
def test_load_coarse_infer_config_defaults_global_path_tracing_disabled() -> None: ...
def test_load_coarse_infer_config_accepts_global_path_tracing_block() -> None: ...
def test_load_coarse_infer_config_rejects_probability_score_type() -> None: ...
def test_load_coarse_infer_config_rejects_negative_max_jump() -> None: ...
def test_load_coarse_infer_config_rejects_zero_min_component_anchors() -> None: ...
def test_load_coarse_infer_config_accepts_dp3d_reserved_algorithm() -> None: ...
```

## 受け入れ条件

- 既存 coarse infer config は block なしで parse できる。
- 新 block は parse できる。
- `score_type` は `log_softmax` 以外を許可しない。
- まだ coarse inference 結果は変わらない。

## 推奨確認コマンド

```bash
pytest packages/seisai-engine/tests/test_fbpick_coarse.py -q
```
